source/code/plugins/npmd_config_lib.rb

```diff
@@ -64,6 +64,56 @@ def getNetMaskString
     class AgentConfigCreator
         public
 
+        # Variables for tracking errors
+        @@agent_ip_drops = 0
+        @@agent_drops = 0
+        @@network_subnet_drops = 0
+        @@network_drops = 0
+        @@rule_subnetpair_drops = 0
+        @@rule_drops = 0
+
+        # Reset error checking
+        def self.resetErrorCheck
+            @@agent_ip_drops = 0
+            @@agent_drops = 0
+            @@network_subnet_drops = 0
+            @@network_drops = 0
+            @@rule_subnetpair_drops = 0
+            @@rule_drops = 0
+        end
+
+        # Generating the error string
+        def self.getErrorSummary
+            _agentIpDrops=""
+            _agentDrops=""
+            _networkSNDrops=""
+            _networkDrops=""
+            _ruleSNPairDrops=""
+            _ruleDrops=""
+
+            if @@agent_ip_drops != 0
+                _agentIpDrops = "Agent IPs=#{@@agent_ip_drops}"
+            end
+            if @@agent_drops != 0
+                _agentDrops= "Agents=#{@@agent_drops}"
+            end
+            if @@network_subnet_drops != 0
+                _networkSNDrops = "Network subnets=#{@@network_subnet_drops}"
```

Comment:
I would suggest to have error strings defined as consts in the beginning instead of having them hardcoded throughout the code in case the they need to be changed or localized.